src/main/java/htsjdk/variant/variantcontext/writer/BCF2Writer.java

```diff
@@ -221,6 +206,43 @@ public void close() {
         super.close();
     }
 
+    @Override
+    public void setVcfHeader(VCFHeader header) {
+        if (isHeaderWritten) {
+            throw new IllegalStateException("Header is already written");
+        }
+        if (isBodyWritten) {
+            throw new IllegalStateException("Body is already written");
+        }
+        // make sure the header is sorted correctly
+        header = new VCFHeader(header.getMetaDataInSortedOrder(), header.getGenotypeSamples());
+
+        this.header = doNotWriteGenotypes ? new VCFHeader(header.getMetaDataInSortedOrder()) : header;
```

Comment:
Do you mean something like this? `this.header = doNotWriteGenotypes ? new VCFHeader(header.getMetaDataInSortedOrder()) : new VCFHeader(header.getMetaDataInSortedOrder(), header.getGenotypeSamples());`